### PR TITLE
expression: fixed pow/power return wrong error messgae when there mor…

### PR DESCRIPTION
**Pull request by zhexuany**
…e than two

arguments.

### CODE REVIEW

**Comment by zhexuany** [comment]
This fixed the issue https://github.com/pingcap/tidb/issues/2848

**Comment by zhexuany** [comment]
@coocood  @zimulala PTAL

**Comment by coocood** [comment]
@zhexuany 
Please add test cases to verify the error code has changed.

**Comment by zhexuany** [comment]
@coocood I do not see any place for testing. The change I made is let query like `select pow(1, 1, 1)` passed into `expression` are. The number of arguments are handled by `baseFunctionClass`.

**Comment by zhexuany** [comment]
@coocood Is this what you want ?

File: parser/parser_test.go [hunk 1]
```diff
@@ -502,6 +502,7 @@ func (s *testParserSuite) TestBuiltin(c *C) {
 	table := []testCase{
 		// for builtin functions
 		{"SELECT POW(1, 2)", true},
+		{"SELECT POW(1, 2, 1)", true}, // illegal number of arguments shall pass too
```
Comment (coocood) [inline parser/parser_test.go]: still pass.

Comment (zhexuany) [inline parser/parser_test.go]: shall pass too is identical with still pass.

Comment (coocood) [inline parser/parser_test.go]: ok

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**